node/node-sdk.md

```diff
@@ -0,0 +1,187 @@
+
+### Example
+```typescript
+export async function main(x: number, y: number) {
+  const msToSleep = await activities.httpGet(`https://calculator.com/add?x=${x}&y={y}`);
+  await new Promise((resolve) => setTimeout(resolve, msToSleep * Math.random()));
+}
+```
+
+### Activity Options
+Activity options are passed in using the `withOptions` method.
+
+```typescript
+export async function main() {
+  const body = await activities.httpGet.withOptions({ retries: 3 }).invoke('http://example.com');
+  console.log(body);
+}
+```
+
+### Activity Type Declarations
+To define type declarations for your workflow's activities, add each activity to your `global.d.ts`.
+```typescript
+/// <reference types="@temporal-sdk/workflow-global" />
+
+import * as declarations from '../../activities';
+
+declare global {
+  namespace activities {
+    // Use the signature of an activity function
+    export var httpGet: Activity<typeof declarations.httpGet>;
+    // In case you can't import you function definition (implmented in another language or simply unavailable in this context),
+    // define the signature manually
+    export var add: Activity<Fn<[number, number], number>>;
+  }
+}
+```
+
+### Registration
+```typescript
+async function main() {
+  const example = 'dist/workflow.js'; // Workflow file generated using the typescript compiler
+  await worker.registerWorkflow(example);
+}
+```
+
+### Limitations
+* The [NodeJS API][nodejs-api] is not available in workflow code as it breaks determinism.
+* Only ES modules can be imported out of the box with V8 isolates, npm uses the CommonJS module implementation.
+    * There are workarounds for this issue - solution TBD.
+    * We might need to implement CommonJS module support: https://nodejs.org/api/modules.html#modules_all_together
+* Submitted workflow code must not use async / await syntax.
+    Using async / await bypasses the injected Promise object in the workflow context and hinders our ability to gain full control over scheduling.
+    To get around this we recommend setting the `tsconfig.json` target to `<=es6` which transforms all async await calls into generators and Promises.
+    Other solutions include using babel [transform regenerator](https://babeljs.io/docs/en/babel-plugin-transform-regenerator) for e.g. when using vanilla JS.
+* `es2015.collection` library should be excluded from `tsconfig.json` because it exports [`WeakMap`][mdn-weakmap] and [`WeakSet`][mdn-weakset].
+    Weak references are non-deterministic since we can't programatically control the V8 garbage collector.
+    [`WeakRef`][v8-weakref] (`v8 >= 8.4` -> `nodejs >= 14.6.0`) suffers from the same issue and should be avoided.
+    `WeakRef`, `WeakMap` and `WeakSet` are all deleted from the global scope.
+
+### Signals
+TBD
+
+### Queries
+TBD
+
+### Logging
+Logging from the workflow can be implemented with `console.log`.
+TBD log format and context
```

Comment:
Does V8 expose the ability to intercept the console log calls? We need it for deduping logging on replay.